src/infer_context.cc

```diff
@@ -214,19 +217,11 @@ InferContext::GetOutputs(const cb::InferResult& infer_result)
   RequestRecord::ResponseOutput output{};
   for (const auto& requested_output : infer_data_.outputs_) {
     std::string data_type{requested_output->Datatype()};
-    const uint8_t* buf{nullptr};
-    size_t byte_size{0};
-    infer_result.RawData(requested_output->Name(), &buf, &byte_size);
+    std::vector<uint8_t> buf{};
+    infer_result.RawData(requested_output->Name(), buf);
 
-    // The first 4 bytes of BYTES data is a 32-bit integer to indicate the size
```

Comment:
So this change here feels a bit clumsy, overall. What is the motivation to change the behavior of this function, and push it down the workflow, aside from trying to reduce copies, maybe?

Reply:
Yeah, so the overhead was because there was a resize here and for larger outputs, it's a big operation.  To avoid the additional overhead, I moved this condition down in the workflow so that we don't need to resize the vector again.

Reply:
When I removed this resize operation, the additional overhead reduced.

Reply:
Yep, okay, makes sense, thanks for the confirmation of my suspicion here.

This is still a sort of API change, but I'd leave this for a later refactor...